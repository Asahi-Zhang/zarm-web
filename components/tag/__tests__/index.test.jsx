import React from 'react';
import { render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Tag from '../index';

describe('Tag', () => {
  it('renders normal Tag correctly', () => {
    const wrapper = render(
      <div className="multi-rows">
        <Tag>default</Tag>
        <Tag theme="primary">primary</Tag>
        <Tag theme="success">success</Tag>
        <Tag theme="warning">warning</Tag>
        <Tag theme="danger">danger</Tag>
      </div>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders radius Tag correctly', () => {
    const wrapper = render(
      <div>
        <Tag>radius</Tag>
        <Tag shape="rect">rect</Tag>
        <Tag shape="round">round</Tag>
      </div>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Tag size correctly ', () => {
    const wrapper = render(
      <div className="multi-rows">
        <Tag>default</Tag>
        <Tag size="large" theme="info">large</Tag>
        <Tag theme="warning">middle</Tag>
        <Tag size="small" theme="success">small</Tag>
        <Tag size="xsmall" theme="danger">xsmall</Tag>
      </div>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('behaves correctly when closing tag', () => {
    const onClose = jest.fn();
    const wrapper = mount(<Tag closable onClose={onClose}>可关闭标签</Tag>);

    wrapper.find('.za-icon').simulate('click');
    expect(onClose).toHaveBeenCalled();
  });
});
