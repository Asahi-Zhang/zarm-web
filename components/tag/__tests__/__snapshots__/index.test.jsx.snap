// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag behaves correctly when click CheckableTag 1`] = `
<CheckableTag
  checked={true}
  onChange={[MockFunction]}
  prefixCls="zw-tag"
  size=""
>
  <div
    className="zw-tag zw-tag--checkable zw-tag--checked"
    onClick={[Function]}
  >
    default
  </div>
</CheckableTag>
`;

exports[`Tag behaves correctly when disabled CheckableTag clicked uncalled onChange 1`] = `
<CheckableTag
  disabled={true}
  onChange={[MockFunction]}
  prefixCls="zw-tag"
  size=""
>
  <div
    className="zw-tag zw-tag--checkable zw-tag--disabled"
    onClick={[Function]}
  >
    default
  </div>
</CheckableTag>
`;

exports[`Tag render CheckableTag correctly 1`] = `
<CheckableTag
  checked={true}
  prefixCls="zw-tag"
  size=""
>
  <div
    className="zw-tag zw-tag--checkable zw-tag--checked"
    onClick={[Function]}
  >
    default
  </div>
</CheckableTag>
`;

exports[`Tag render CheckableTag size correctly 1`] = `
<div
  class="multi-rows"
>
  <div
    class="zw-tag zw-tag--checkable zw-tag--lg"
  >
    lg
  </div>
  <div
    class="zw-tag zw-tag--checkable zw-tag--md"
  >
    md
  </div>
  <div
    class="zw-tag zw-tag--checkable zw-tag--sm"
  >
    lg
  </div>
  <div
    class="zw-tag zw-tag--checkable zw-tag--xs"
  >
    lg
  </div>
</div>
`;

exports[`Tag renders Tag bordered correctly 1`] = `
<div
  class="multi-rows"
>
  <div
    class="zw-tag zw-tag--red zw-tag--md zw-tag--radius"
  >
    normal
  </div>
  <div
    class="zw-tag zw-tag--green zw-tag--md zw-tag--radius zw-tag--unborder"
    style="color:#fff;border:hidden"
  >
    unbordered
  </div>
</div>
`;

exports[`Tag renders Tag color correctly 1`] = `
<div
  class="multi-rows"
>
  <div
    class="zw-tag zw-tag--red zw-tag--md zw-tag--radius"
  >
    normal
  </div>
</div>
`;

exports[`Tag renders Tag shape correctly 1`] = `
<div>
  <div
    class="zw-tag zw-tag--md zw-tag--radius"
  >
    radius
  </div>
  <div
    class="zw-tag zw-tag--md zw-tag--rect"
  >
    rect
  </div>
  <div
    class="zw-tag zw-tag--md zw-tag--round"
  >
    round
  </div>
</div>
`;

exports[`Tag renders Tag size correctly  1`] = `
<div
  class="multi-rows"
>
  <div
    class="zw-tag zw-tag--md zw-tag--radius"
  >
    default
  </div>
  <div
    class="zw-tag zw-tag--lg zw-tag--radius"
  >
    lg
  </div>
  <div
    class="zw-tag zw-tag--md zw-tag--radius"
  >
    md
  </div>
  <div
    class="zw-tag zw-tag--sm zw-tag--radius"
  >
    sm
  </div>
  <div
    class="zw-tag zw-tag--xs zw-tag--radius"
  >
    xs
  </div>
</div>
`;

exports[`Tag renders normal Tag correctly 1`] = `
<div
  class="multi-rows"
>
  <div
    class="zw-tag zw-tag--md zw-tag--radius"
  >
    normal
  </div>
</div>
`;
